### ISSUE TITLE
Pytorch reimplementation

### ISSUE THREAD

**Issue by zhangmozhe** [tranluan/Nonlinear_Face_3DMM#37]
Hi, I really like this work. I just wonder whether you have the plan to reimplement it using Pytorch? Or release the pytorch pre-trained model?

**Comment by HoiM** [comment]
I am currently working on this project with PyTorch and PyTorch3D. Things are almost done except the settings of the camera so that the 8 transformation parameters can work with the rendered image. Now I hope the author could provide the parameters of the camera.

**Comment by invisprints** [comment]
@HoiM Hello, any progress？

**Comment by HoiM** [comment]
> @HoiM Hello, any progress？

Got no reply from the author...

**Comment by invisprints** [comment]
@HoiM I am confused about the parameters of the camera you said. are they different from camera projection parameters？In the paper, it said "the camera projection parameters estimated by the encoder"

**Comment by chaoshiedwin** [comment]
> I am currently working on this project with PyTorch and PyTorch3D. Things are almost done except the settings of the camera so that the 8 transformation parameters can work with the rendered image. Now I hope the author could provide the parameters of the camera.

Hi, did you find that the z-buffer kernel is not differentiable in the code? How can you resovle this problem with pytorch?

**Comment by HoiM** [comment]
@invisprints Maybe I was confused. The camera model used in the paper is weak perspective projection camera model. Based on my understanding, the 8 parameters can directly convert a 3D point to a plane. While in my implementation, I used the perspective projection camera model, where field of view, location of the camera in the world system, and the focal length must be set. Maybe the 8 parameters are already enough. 

Please feel free to correct me if anything I said was wrong.

**Comment by HoiM** [comment]
@chaoshiedwin I switched to another work and successfully implemented the rendering process. Here the camera model is weak perspective model. But I used ```OpenGLPerspectiveCamera``` in PyTorch3D, where Z-buffer is not needed to be coded by hand. 

Moreover, based on my understanding, z-buffer is more of like a mask where back-propagation is not needed.
